core/storage/base_model/gae_models.py

```diff
@@ -259,17 +298,18 @@ def get_multi(cls, entity_ids, include_deleted=False):
             else:
                 none_argument_indices.append(index)
 
-        entities = datastore_services.get_multi(entity_keys)
+        entities = datastore_services.get_multi(entity_keys) # type: List[Optional[SELF_BASE_MODEL]]
         for index in none_argument_indices:
             entities.insert(index, None)
 
         if not include_deleted:
             for i in python_utils.RANGE(len(entities)):
-                if entities[i] and entities[i].deleted:
+                if entities[i] and entities[i].deleted: # type: ignore[union-attr]
```

Comment:
Generally, any ignore statement (other than no-untyped-call) needs an explanation.

Reply:
Here the point is that MyPy throws the error entities[i] may have value None and entities[i].deleted will not exist in this case. But mypy being a static type checker is not able to detect that here we already checked the None case. That's why

Reply:
I thought we already had a mechanism for that -- either casting or asserts.

Reply:
But here casting is not a good idea. We ourselver are not sure that the value is none or None. So we ourselves are checking that using `if entities[i] and entities[i].deleted:`

Reply:
Then refactor the code such that you check whether entities[i] exists, do the assertion for mypy, and then check entities[i].deleted.

Reply:
Tried to refactor. Was unsuccessful. Moved the discussion to the doc. Thanks!

Reply:
Fixed